iotdb/src/main/java/org/apache/iotdb/db/writelog/transfer/PhysicalPlanLogTransfer.java

```diff
@@ -20,28 +20,41 @@
 
 import java.io.IOException;
 import java.nio.BufferOverflowException;
+import org.apache.iotdb.db.auth.AuthException;
 import org.apache.iotdb.db.exception.WALOverSizedException;
 import org.apache.iotdb.db.qp.physical.PhysicalPlan;
+import org.apache.iotdb.db.qp.physical.crud.DeletePlan;
+import org.apache.iotdb.db.qp.physical.crud.InsertPlan;
+import org.apache.iotdb.db.qp.physical.crud.UpdatePlan;
+import org.apache.iotdb.db.qp.physical.sys.AuthorPlan;
+import org.apache.iotdb.db.qp.physical.sys.LoadDataPlan;
+import org.apache.iotdb.db.qp.physical.sys.MetadataPlan;
+import org.apache.iotdb.db.qp.physical.sys.PropertyPlan;
 
 public class PhysicalPlanLogTransfer {
 
-  private PhysicalPlanLogTransfer(){}
+  private PhysicalPlanLogTransfer() {
+  }
 
   public static byte[] operatorToLog(PhysicalPlan plan) throws IOException {
     Codec<PhysicalPlan> codec;
-    switch (plan.getOperatorType()) {
-      case INSERT:
-        codec = (Codec<PhysicalPlan>) PhysicalPlanCodec.fromOpcode(SystemLogOperator.INSERT).codec;
-        break;
-      case UPDATE:
-        codec = (Codec<PhysicalPlan>) PhysicalPlanCodec.fromOpcode(SystemLogOperator.UPDATE).codec;
-        break;
-      case DELETE:
-        codec = (Codec<PhysicalPlan>) PhysicalPlanCodec.fromOpcode(SystemLogOperator.DELETE).codec;
-        break;
-      default:
-        throw new UnsupportedOperationException(
-            "SystemLogOperator given is not supported. " + plan.getOperatorType());
+    if (plan instanceof InsertPlan) {
```

Comment:
why.. do you change a switch-case to if-else?

Reply:
Because multi OperatorTypes may be the same type of physical plan, so I think it's more understandable to distinguish in this way.

Reply:
I make a test and switch method is more efficient, so I change to switch.